pvlib/iam.py

```diff
@@ -274,6 +274,111 @@ def martin_ruiz(aoi, a_r=0.16):
     return iam
 
 
+def martin_ruiz_diffuse(slope, a_r=0.16, c1=0.4244, c2=None):
+    '''
+    Determine the incidence angle modifiers (iam) for diffuse sky and
+    ground-reflected irradiance using the Martin and Ruiz incident angle model.
+
+    Parameters
+    ----------
+    slope : numeric, degrees
+        The slope or tilt angle of the module, where 0 is horizontal.
+
+    a_r : numeric
+        The angular losses coefficient described in equation 3 of [1].
+        This is an empirical dimensionless parameter. Values of a_r are
+        generally on the order of 0.08 to 0.25 for flat-plate PV modules.
+        a_r must be a positive numeric scalar.
+
+    c1, c2 : numeric scalar
+        Two fitting parameters for the expressions that approximate the
+        integral of diffuse irradiance coming from different directions.
+        c1 is contant at 4 / 3 / pi (0.4244) and c2 varies with a_r.  The
+        calculation of c2 from a_r is according to [3].
+
+    Returns
+    -------
+    iam_sky : numeric
+        The incident angle modifier(s) for sky diffuse
+
+    iam_ground : numeric
+        The incident angle modifier(s) for ground-reflected diffuse
+
+    Notes
+    -----
+    Sky and ground modifiers are complementary: iam_sky for slope = 30 is
+    equal to iam_ground for slope = 180 - 30.  For vertical surfaces,
+    slope = 90, the two factors are equal.
+
+    References
+    ----------
+    [1] N. Martin and J. M. Ruiz, "Calculation of the PV modules angular
+    losses under field conditions by means of an analytical model", Solar
+    Energy Materials & Solar Cells, vol. 70, pp. 25-38, 2001.
+
+    [2] N. Martin and J. M. Ruiz, "Corrigendum to 'Calculation of the PV
+    modules angular losses under field conditions by means of an
+    analytical model'", Solar Energy Materials & Solar Cells, vol. 110,
+    pp. 154, 2013.
+
+    [3] "IEC 61853-3 Photovoltaic (PV) module performance testing and energy
+    rating - Part 3: Energy rating of PV modules". IEC, Geneva, 2018.
+
+    See Also
+    --------
+    iam.martin_ruiz
+    iam.physical
+    iam.ashrae
+    iam.interp
+    iam.sapm
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Oct. 2019
+
+    if isinstance(slope, pd.Series):
+        out_index = slope.index
+    else:
+        out_index = None
+
+    slope = np.asanyarray(slope)
+
+    with np.errstate(invalid='ignore'):
```

Comment:
I'm not following the purpose of the slope manipulations - what values of slope are anticipated here?

Reply:
Any value is allowed.  Maybe I got carried away.

Reply:
According to the description in the docstring, 0 <= `surface_tilt` <= 180. I don't think this function should adjust `surface_tilt` for negative values or values greater than 360. I'd prefer to raise `ValueError` and let the user figure out where the incorrect tilt values came from.

Reply:
I'm generally opposed to input checking on the grounds that it's a huge amount of work to maintain and it gives users a false sense of security (they prevented me from doing something stupid in function a, so I'm not going to worry about doing something stupid in function b). Garbage in, garbage out. I'd relax my position for higher-level wrapper functions/methods.

Reply:
To docstring doesn't say " 0 <= surface_tilt <= 180 ", does it? In my own code I interpret azimuth 0 with slope -30 to the the same as azimuth 180, slope 30.  Sometimes that's convenient.
Anyway, I'll make it do whatever the majority/authority would like, but it's not clear what that is at the moment.

Reply:
It doesn't and perhaps it should. It says "degrees from horizontal" and we don't define a positive rotation direction , so in my view it is simply an angle without a rotation direction. Your example has an implicit direction (positive is toward the azimuth). 

The [styel guide](https://pvlib-python.readthedocs.io/en/latest/variables_style_rules.html) is less clear on this point.

Reply:
How many functions would need to gain these incantations to keep things consistent through the library?

Reply:
@wholmgren the manipulations to ensure `surface_tilt` is between 0 and 180?  

To clarify my point of view, 0 < surface_tilt < 180 is how I've seen the quantity used, and I think we can state that in the docstring and omit the manipulations and value checking.

Reply:
> the manipulations to ensure surface_tilt is between 0 and 180?

Yes. 

> To clarify my point of view, 0 < surface_tilt < 180 is how I've seen the quantity used, and I think we can state that in the docstring and omit the manipulations and value checking.

This is my preference too.

Reply:
Ok, I will remove all that stuff.  Shall I remove the test on `a_r` also?

Reply:
> Shall I remove the test on a_r also?

I don't have a preference. The docstring states the restriction on the value.